modules/features/podcasts/src/test/java/au/com/shiftyjelly/pocketcasts/podcasts/view/folders/SuggestedFoldersViewModelTest.kt

```diff
@@ -0,0 +1,198 @@
+package au.com.shiftyjelly.pocketcasts.podcasts.view.folders
+
+import au.com.shiftyjelly.pocketcasts.analytics.AnalyticsEvent
+import au.com.shiftyjelly.pocketcasts.analytics.AnalyticsTracker
+import au.com.shiftyjelly.pocketcasts.models.to.SignInState
+import au.com.shiftyjelly.pocketcasts.models.to.SubscriptionStatus
+import au.com.shiftyjelly.pocketcasts.repositories.podcast.FolderManager
+import au.com.shiftyjelly.pocketcasts.repositories.podcast.PodcastManager
+import au.com.shiftyjelly.pocketcasts.repositories.podcast.SuggestedFoldersManager
+import au.com.shiftyjelly.pocketcasts.repositories.user.UserManager
+import au.com.shiftyjelly.pocketcasts.sharedtest.MainCoroutineRule
+import io.reactivex.Flowable
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.test.advanceUntilIdle
+import kotlinx.coroutines.test.runTest
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.Mock
+import org.mockito.Mockito.reset
+import org.mockito.MockitoAnnotations
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import au.com.shiftyjelly.pocketcasts.models.entity.SuggestedFolder as DbSuggestedFolder
+
+@ExperimentalCoroutinesApi
+class SuggestedFoldersViewModelTest {
+
+    @get:Rule
+    val coroutineRule = MainCoroutineRule()
+
+    private lateinit var viewModel: SuggestedFoldersViewModel
+
+    @Mock
+    private lateinit var folderManager: FolderManager
+
+    @Mock
+    private lateinit var suggestedFoldersManager: SuggestedFoldersManager
+
+    @Mock
+    private lateinit var suggestedFoldersPopupPolicy: SuggestedFoldersPopupPolicy
+
+    @Mock
+    private lateinit var podcastManager: PodcastManager
+
+    @Mock
+    private lateinit var userManager: UserManager
+
+    @Mock
+    private lateinit var analyticsTracker: AnalyticsTracker
+
+    @Mock
+    private lateinit var source: SuggestedFoldersFragment.Source
+
+    private val folderCount = 1
+    private val dbSuggestedFolders = listOf(DbSuggestedFolder("Folder1", "podcast1"))
+
+    @Before
+    fun setup() {
+        MockitoAnnotations.openMocks(this)
+    }
+
+    @After
+    fun tearDown() {
+        reset(
+            folderManager,
+            suggestedFoldersManager,
+            suggestedFoldersPopupPolicy,
+            podcastManager,
+            userManager,
+            analyticsTracker,
+            source,
+        )
+    }
+
+    @Test
+    fun `popup was marked as dismissed`() = coroutineRule.testScope.runTest {
```

Comment:
```suggestion
    fun `popup was marked as dismissed`() = runTest {
```
I don't fully understand these test scopes. Do these need to have the testScope in them? We don't seem to have done this anywhere else in the code.

Reply:
This approach provides tighter control over timing, dispatcher configuration, and cleanup. It was another attempt to fix the flaky tests. Since I also added `advanceUntilIdle()`, which should help address the issue, I'll try using runTest to see if `advanceUntilIdle()` alone fixes the problem